php/class-photonfill.php

```diff
@@ -454,42 +479,52 @@ public function create_url_image_object( $img_url, $current_size, $args = array(
 						$default = ( ! empty( $img_size['default'] ) ) ? true : false;
 						$current_w = empty( $img_size['width'] ) ? 0 : $img_size['width'];
 						$current_h = empty( $img_size['height'] ) ? 0 : $img_size['height'];
-						$transform_args = array(
+						$transform_args = apply_filters( 'photonfill_pre_transform_args', array(
 							'callback' => ( isset( $img_size['callback'] ) ) ? $img_size['callback'] : null,
 							'crop' => ( isset( $img_size['crop'] ) ) ? $img_size['crop'] : true,
 							'breakpoint' => $breakpoint,
 							'image_size' => $current_size,
 							'width' => $current_w,
 							'height' => $current_h,
 							'quality' => ( isset( $img_size['quality'] ) ) ? $img_size['quality'] : null,
-						);
+						), 'external_url', $current_size, $args );
 						$this->transform->setup( $transform_args );
 						$img_src = $this->get_url_img_src( $img_url, array( $current_w, $current_h ), $default );
-						$image_sizes[ $breakpoint ] = array( 'size' => $this->breakpoints[ $breakpoint ], 'src' => $img_src );
+						$image_sizes[ $breakpoint ] = array(
+							'size' => $this->breakpoints[ $breakpoint ],
+							'src' => $img_src,
+						);
 					}
 				} elseif ( is_array( $current_size ) ) {
 					// If our size in an array of ints parse it differently.
 					foreach ( $this->breakpoints as $breakpoint => $breakpoint_widths ) {
 						$breakpoint_width = $this->get_breakpoint_width( $breakpoint );
 						$breakpoint_height = ( ! empty( $breakpoint_widths['height'] ) ) ? $breakpoint_widths['height'] : 9999;
 						$new_size = wp_constrain_dimensions( $current_size[0], $current_size[1], $breakpoint_width, $breakpoint_height );
-						$transform_args = array(
+						$transform_args = apply_filters( 'photonfill_pre_transform_args', array(
 							'callback' => ( isset( $img_size['callback'] ) ) ? $img_size['callback'] : null,
 							'crop' => ( isset( $breakpoint_widths['crop'] ) ) ? $breakpoint_widths['crop'] : true,
 							'breakpoint' => $breakpoint,
 							'image_size' => 'full',
 							'width' => $new_size[0],
 							'height' => $new_size[1],
 							'quality' => ( isset( $breakpoint_widths['quality'] ) ) ? $breakpoint_widths['quality'] : null,
-						);
+						), 'external_url', $current_size, $args );
 						$this->transform->setup( $transform_args );
 						$img_src = $this->get_url_img_src( $img_url, $new_size );
-						$image_sizes[ $breakpoint ] = array( 'size' => $this->breakpoints[ $breakpoint ], 'src' => $img_src );
+						$image_sizes[ $breakpoint ] = array(
+							'size' => $this->breakpoints[ $breakpoint ],
+							'src' => $img_src,
+						);
 					}
-				}
+				} // End if().
 
-				return array( 'id' => 'external_url', 'sizes' => $image_sizes, 'args' => $args );
-			}
+				return array(
+					'id' => 'external_url',
```

Comment:
... and finally here. Does it make more sense to set this as a variable like `$id` somewhere above rather than repeat the hardcoded text multiple times? Seems easier for future maintenance.